src/__tests__/gatsby-node.tests.js

```diff
@@ -1,27 +1,181 @@
 import {
-  urlWithoutParameters,
+  fetchPlone,
+  fetchUrl,
   headersWithToken,
+  logging,
+  normalizeData,
   normalizePath,
-} from '../normalize';
+  parentId,
+  urlWithoutParameters,
+} from '../utils';
 
-test('url without parameters', () => {
+test('urlWithoutparameters', () => {
   expect(
     urlWithoutParameters('https://example.com/search?q=examplequery')
   ).toBe('https://example.com/search');
 });
 
-test('headers with token', () => {
+test('headersWithToken', () => {
   expect(headersWithToken({ accept: 'application/json' }, 'token')).toEqual({
     accept: 'application/json',
     Authorization: 'Bearer token',
   });
 });
 
-test('normalize path', () => {
+test('normalizePath', () => {
   expect(normalizePath('foo')).toBe('/foo/');
   expect(normalizePath('/foo')).toBe('/foo/');
   expect(normalizePath('foo/')).toBe('/foo/');
   expect(normalizePath('/foo/')).toBe('/foo/');
   expect(normalizePath('//foo//')).toBe('/foo/');
   expect(normalizePath(undefined)).toBe('/');
 });
+
+test('parentId', () => {
+  expect(parentId('http://localhost:8080/foo/bar')).toBe(
+    'http://localhost:8080/foo'
+  );
+  expect(parentId('http://localhost:8080/foo')).toBe('http://localhost:8080');
+  expect(parentId('http://localhost:8080/')).toBe('http://localhost:8080');
+  expect(parentId('http://localhost:8080')).toBe('http://localhost:8080');
+});
+
+const http = {
+  get: async (url, headers, params) =>
+    url !== 'next'
+      ? {
+          data: {
+            ...headers,
+            batching: { next: 'next' },
+            items: [1],
+          },
+        }
+      : {
+          data: {
+            ...headers,
+            items: [2],
+            batching: {},
+          },
+        },
+};
+
+test('fetchUrl', async () => {
+  const data = await fetchUrl('url', 'token', {}, http);
+  expect(data.headers).toEqual({
+    Accept: 'application/json',
+    Authorization: 'Bearer token',
+  });
+});
+
+test('fetchPlone', async () => {
+  const data = await fetchPlone('url', 'token', {}, http);
+  expect(data.items).toEqual([1, 2]);
```

Comment:
Whenever you get time, could please brief me on how this test works? It seems quite baffling to me reading it!

Reply:
It's a bit similar how logging is tested...
 
fetchUrl takes one extra parameter that defaults to axios library. That's only there to allow tests to pass mock version of that library to capture test call.

In this particular test the passed mock returns a response with `data.batching.next` be default and when it gets called with the url in `data.batching.next` it returns alternative value without batching. fetchPlone includes your code to loop until all batching urls are fetched.